platform/ui/src/hooks/useMedia.js

```diff
@@ -17,36 +33,37 @@ import { useState, useEffect } from 'react';
  * @returns
  */
 function useMedia(queries, values, defaultValue) {
-  // Array containing a media query list for each query
-  const mediaQueryLists = queries.map(q => window.matchMedia(q));
+  const memoizedQueryMap = useMemo(() => getMediaQueryMap(queries), [queries]);
+  const memoizedValue = useMemo(
+    () => getValue(memoizedQueryMap, values, defaultValue),
+    [mediaQueryLists, values]
+  );
+  // State and setter for matched value
+  const [mediaQueryLists, setMediaQueryLists] = useState(memoizedQueryMap);
+  const [value, setValue] = useState(memoizedValue);
+
+  const mediaQueryListener = () => {
+    const nextValue = getValue(mediaQueryLists, values, defaultValue);
+    setValue(nextValue);
+  };
 
-  // Function that gets value based on matching media query
-  const getValue = () => {
-    // Get index of first media query that matches
-    const index = mediaQueryLists.findIndex(mql => mql.matches);
+  const addMediaQueryListeners = mediaQueryLists => {
+    // Set a listener for each media query with above mediaQueryListener as callback.
+    mediaQueryLists.forEach(mql => mql.addListener(mediaQueryListener));
+  };
 
-    // Return related value or defaultValue if none
-    return typeof values[index] !== 'undefined' ? values[index] : defaultValue;
+  const removeMediaQueryListeners = (mediaQueryLists = []) => {
+    mediaQueryLists.forEach(mql => mql.removeListener(mediaQueryListener));
   };
 
-  // State and setter for matched value
-  const [value, setValue] = useState(getValue);
-
-  useEffect(
-    () => {
-      // Event listener callback
-      // Note: By defining getValue outside of useEffect we ensure that it has ...
-      // ... current values of hook args (as this hook callback is created once on mount).
-      const handler = () => setValue(getValue);
-
-      // Set a listener for each media query with above handler as callback.
-      mediaQueryLists.forEach(mql => mql.addListener(handler));
-
-      // Remove listeners on cleanup
-      return () => mediaQueryLists.forEach(mql => mql.removeListener(handler));
-    },
-    [] // Empty array ensures effect is only run on mount and unmount
-  );
+  useEffect(() => {
+    addMediaQueryListeners(mediaQueryLists);
+  }, [mediaQueryLists]);
+
+  useEffect(() => {
+    // Remove listeners on cleanup
+    return () => removeMediaQueryListeners(mediaQueryLists);
+  }, []);
```

Comment:
`[]` means this is only run once, but `[mediaQueryLists]` above means the effect is run each time `mediaQueryLists` is modified.

I'm not 100% sure I follow the logic of this refactor -- I may need a walkthrough. I've only used `useMemo` once or twice, so I'm likely missing something obvious.